mail_edit/models/compose_message.py

```diff
@@ -0,0 +1,91 @@
+# -*- coding: utf-8 -*-
+# © 2016 Sunflower IT (http://sunflowerweb.nl)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openerp import api, fields, models
+
+
+class MailEditComposeMessage(models.TransientModel):
+    _name = "mail_edit.compose_message"
+    _inherits = {"mail.compose.message": "original_wizard_id"}
+
+    @api.model
+    def default_get(self, fields):
+        result = self.original_wizard_id.default_get(fields)
+
+        if "subject" in result and "default_subject" in self.env.context:
+            result["subject"] = self.env.context["default_subject"]
+
+        result["message_id"] = self.env.context["default_message_id"]
+        result["move_mail"] = True
+        return result
+
+    @api.model
+    def _get_model_selection(self):
+        """Get allowed models and their names."""
+        model_objs = self.env["res.request.link"].search(
+            [("mail_edit", "=", True)],
+            order="name")
+        return [(m.object, m.name) for m in model_objs]
+
+    @api.one
+    @api.onchange("destination_object_id")
+    def change_destination_object(self):
+        """Update some fields for the new message."""
+        if self.destination_object_id:
+            self.model = self.destination_object_id._name
+            self.res_id = self.destination_object_id.id
+
+            model_name = (self.env["ir.model"]
+                          .search([("model", "=", self.model)])
+                          .name)
```

Comment:
this is not review but personal taste: I guess you're not happy with this formatting either, What I'd have done in this case is

```python
            model_name = self.env["ir.model"].search([
                ("model", "=", self.model)
            ]).name
```

to my mind, that tells me faster what this is about, but as said, no review, friendly advise